nep-11.mediawiki

```diff
@@ -0,0 +1,251 @@
+<pre>
+  NEP: 11
+  Title: Non-Fungible Token Standard
+  Author: Joe Stewart <hal0x2328@splyse.tech>, Shane Mann <fetter@splyse.tech>, Wyatt Mufson <wyatt@ryu.games>, Erik Zhang <erik@neo.org>
+  Type: Standard
+  Status: Accepted
+  Created: 2018-04-18
+</pre>
+
+==Abstract==
+
+This NEP defines a standard non-fungible token system for the NEO Smart Economy.
+
+==Motivation==
+
+NFTs are required to track, exchange and enforce ownership of digital assets.
+A non-fungible token (NFT) can be thought of like a property deed - each one is unique and carries some non-mutable information (e.g. the physical address of the property).
+Other information, such as the owner of the property, can be changed.
+Also, we provide built-in optional divisibility within each non-fungible asset.
+This allows for high value objects to be tokenized more effectively.
+
+==Specification==
+
+===Common methods===
+
+====symbol====
+
+<pre>
+public static string symbol()
+</pre>
+
+Returns a short string symbol of the token managed in this contract. e.g. <code>"MNFT"</code>.
+This symbol SHOULD be short (3-8 characters is recommended), with no whitespace characters or new-lines and SHOULD be limited to the uppercase latin alphabet (i.e. the 26 letters used in English).
+
+This method MUST always return the same value every time it is invoked.
+
+====decimals====
+
+<pre>
+public static byte decimals()
+</pre>
+
+Returns the number of decimals used by the token - e.g. <code>8</code>, means to divide the token amount by <code>100,000,000</code> to get its user representation.
+
+If the token managed in this contract is indivisible, the function SHOULD return <code>0</code>.
+
+If this method returns <code>0</code>, the "Non-divisible NFT methods" must be implemented.
+Otherwise, the "Divisible NFT methods" must be implemented.
+
+This method MUST always return the same value every time it is invoked.
+
+====totalSupply====
+
+<pre>
+public static BigInteger totalSupply()
+</pre>
+
+Returns the total token supply currently in the system.
+
+====balanceOf====
+
+<pre>
+public static BigInteger balanceOf(byte[] owner)
```

Comment:
This `balanceOf` is a little bit strange.
What is this total amount of NFTs? Since we do not describe the tokensID this info may become misleading.

Reply:
See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md#specification